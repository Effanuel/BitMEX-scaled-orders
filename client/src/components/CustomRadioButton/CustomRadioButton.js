import React from "react";
import { FormCheck } from "react-bootstrap";
import styles from "./CustomRadioButton.module.css";

const CustomRadioButton = ({ label, type, name, defaultChecked }) => {
  return (
    <div className={styles.myLabel}>
      <FormCheck
        custom
        name={name}
        label={label}
        type={type}
        id={label}
        value={label}
        defaultChecked={defaultChecked}
      />
    </div>
  );
};

export default CustomRadioButton;
