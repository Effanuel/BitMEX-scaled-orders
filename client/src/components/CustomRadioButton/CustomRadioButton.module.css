.myLabel {
  color: white;
  font-size: 12px;
}
