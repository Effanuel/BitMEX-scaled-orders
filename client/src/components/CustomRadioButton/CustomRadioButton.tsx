import React from 'react';
import {Radio, FormControlLabel, RadioGroup} from '@material-ui/core';
import {makeStyles} from '@material-ui/core/styles';
import {SIDE} from 'redux/api/bitmex/types';

const useStyles = makeStyles(() => ({
  root: {
    '& .MuiFormControlLabel-label': {color: 'rgba(255, 255, 255, 0.6)', fontSize: '14px', margin: '0px'},
    '& .MuiIconButton-root': {color: '#4caf50', padding: '2px'},
    margin: '0px',
  },
  radioGroup: {
    marginTop: '15px',
  },
}));

interface CustomRadioButtonProps {
  id: string;
  value: string;
  label: React.ReactNode;
  checked?: boolean;
  labelPlacement?: 'end' | 'start' | 'top' | 'bottom';
}

const controlButton = <Radio size="small" />;

export function CustomRadioButton(props: CustomRadioButtonProps) {
  const {root} = useStyles();

  return <FormControlLabel labelPlacement={'end'} className={root} control={controlButton} {...props} />;
}

interface SideRadioButtonsProps {
  testID?: string;
  onChangeRadio: (value: SIDE) => void;
  side: SIDE;
}

export function SideRadioButtons({onChangeRadio, side, testID}: SideRadioButtonsProps) {
  const {radioGroup} = useStyles();
  const emitRadioChange = React.useCallback((_, value: string) => onChangeRadio(value as SIDE), [onChangeRadio]);
  return (
    <RadioGroup
      data-test-id={testID}
      aria-label="position"
      name="side"
      value={side}
      onChange={emitRadioChange}
      className={radioGroup}
    >
      <CustomRadioButton id="scaled_side_sell" label="Sell" value="Sell" />
      <CustomRadioButton id="scaled_side_buy" label="Buy" value="Buy" />
    </RadioGroup>
  );
}
