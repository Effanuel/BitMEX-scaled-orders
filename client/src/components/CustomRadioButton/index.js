import React from "react";
import PropTypes from "prop-types";
import { FormCheck } from "react-bootstrap";
import styles from "./CustomRadioButton.module.css";

const CustomRadioButton = ({ label, type, name, defaultChecked }) => {
  return (
    <div className={styles.myLabel}>
      <FormCheck
        custom
        name={name}
        label={label}
        type={type}
        id={label}
        value={label}
        defaultChecked={defaultChecked}
        className={styles.noselect}
      />
    </div>
  );
};

CustomRadioButton.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  type: PropTypes.oneOf(["radio", "checkbox"]),
  value: PropTypes.string,
  defaultChecked: PropTypes.bool
};

export { CustomRadioButton };
