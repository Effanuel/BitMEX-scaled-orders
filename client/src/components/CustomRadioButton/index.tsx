import React from "react";
// COMPONENTS
import { FormCheck } from "react-bootstrap";
// UTILS
import styles from "./styles.module.css";

type Props = {
  label: string;
  type: "radio" | "checkbox";
  name: string;
  defaultChecked?: boolean;
};

function CustomRadioButton({ label, type, defaultChecked, name }: Props) {
  return (
    <div className={styles.label}>
      <FormCheck
        custom
        name={name}
        label={label}
        type={type}
        id={label}
        value={label}
        defaultChecked={defaultChecked}
        className={styles.label__noselect}
      />
    </div>
  );
}

export { CustomRadioButton };
