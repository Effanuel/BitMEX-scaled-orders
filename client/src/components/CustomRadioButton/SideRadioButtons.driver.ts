import {ComponentDriver} from 'tests/fuel/drivers';
import {SIDE} from 'util/BitMEX-types';
import {SideRadioButtons, SideRadioButtonsProps} from './CustomRadioButton';

export default class SideRadioButtonsDriver extends ComponentDriver<SideRadioButtonsProps> {
  constructor() {
    super(SideRadioButtons);
  }
  withDefaultProps(props?: Partial<SideRadioButtonsProps>) {
    const defaultProps: SideRadioButtonsProps = {
      testID: 'default:SideRadioButtons',
      onChangeRadio: jest.fn(),
      side: SIDE.BUY,
    };
    return this.setProps({...defaultProps, ...props});
  }

  getInputValue() {
    const testID = this.props.testID;
    console.log(this.props);
    // console.log(this.getByID(this.props.testID!));
    if (testID) {
      console.log(this.getByID(testID)!.props, '-==-=-=-=-=');
      //   return
    }
  }

  setInputValue(value: string) {
    console.log(this.props);
    // const node = this.getElement(this.props!['data-test-id']!);
    // node.props.onChange({target: {value, name: 'side'}});
    // node.props.value = value;
    //   node.props.value = value;
    // return this;
  }
}
