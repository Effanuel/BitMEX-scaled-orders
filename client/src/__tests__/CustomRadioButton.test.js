import React from "react";
import { shallow } from "enzyme";
import CustomRadioButton from "../components/CustomRadioButton/CustomRadioButton";

const props = {
  label: "Sell",
  type: "radio",
  name: "side",
  defaultChecked: false
};

const component = shallow(<CustomRadioButton {...props} />);

describe("CustomRadioButton component", () => {
  it("renders without crashing given the required props", () => {
    expect(component).toMatchSnapshot();
  });
});
