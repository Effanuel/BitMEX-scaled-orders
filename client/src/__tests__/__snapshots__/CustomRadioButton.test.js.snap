// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomRadioButton component renders without crashing given the required props 1`] = `ShallowWrapper {}`;
